### DISCUSSION TITLE
Question about custom translations

### DISCUSSION THREAD

**Discussion by arno1979** [kimai/kimai#4057] [Questions & Answers] [answered]
I am currently working on a custom PDF export template.
In this template I want to use some custom language variables.
In the export documentary it says:
> You can translate the button for your template, by adding its name to the export translation file, eg. translations/export.en.xlf

If I add my variables to one of these files won't they be overwritten on the next update?
Is it possible to add my own distinct language files containing my translations and so making the process safe for updates?

Thanks
Arno

**Comment by kevinpapst** [comment]
You should be able to create your own file, e.g. `translations/foo.en.xlf`. Rebuild cache afterwards.
Read the Symfony documentation to find out more.

But usually it is enough to make a switch in your template:
```twig
{% set my_key = 'Mein Tee' %}
{% if invoice['invoice.language'] == 'en }}
    {% set my_key = 'My tea' %}
{% endif %}
```

**Reply by arno1979** [reply]
I tried creating my own file but the translation works only if I put the new translations in the original files. In this case these files are "export.de.xlf" (to display the correct template name in the PDF export drop down menu) and "messages.de.xlf" (for adding custom expressions for use inside the actual template).

The Symfony documentation wasn't very helpful, but I didn't have that much time yet to look through all the chapters dealing with translations.

**Reply by kevinpapst** [reply, marked as answer]
Template name is snake-cased in the UI.
So "foo-bar.pdf.twig" becomes "Foo Bar". Should be enough... 

You can translate from other files via `{{ 'key'|trans({}, 'foo') }}`

**Reply by arno1979** [reply]
Great, that works!
Thanks!
